plugins/dynamic-few-shot-lambda/src/GENAIIDP-dynamic-few-shot.py

```diff
@@ -0,0 +1,416 @@
+        extraction_config = config.get("extraction", {})
+        logger.info(f"=== EXTRACTION CONFIG ===")
+        logger.info(f"Model: {extraction_config.get('model', 'Not specified')}")
+        logger.info(f"Temperature: {extraction_config.get('temperature', 'Not specified')}")
+        logger.info(f"Max tokens: {extraction_config.get('max_tokens', 'Not specified')}")
+        logger.info(f"Custom Lambda ARN: {extraction_config.get('custom_prompt_lambda_arn', 'Not specified')}")
+        
+        # Default system prompt from config
+        default_system_prompt = config.get("extraction", {}).get("system_prompt", "")
+        logger.info(f"Default system prompt length: {len(default_system_prompt)} characters")
+        default_task_prompt = config.get("extraction", {}).get("task_prompt", "")
+        logger.info(f"Default task prompt length: {len(default_task_prompt)} characters")
+
+        logger.info(f"=== HANDLE INPUT DOCUMENT ===")
+
+        # Handle input document
+        result = _handle_input_document(placeholders, default_system_prompt, default_task_prompt)
+
+        # Log complete output structure
+        logger.info(f"=== OUTPUT ANALYSIS ===")
+        logger.info(f"Output keys: {list(result.keys())}")
+        logger.info(f"System prompt length: {len(result.get('system_prompt', ''))}")
+        logger.info(f"System prompt (first 200 chars): {result.get('system_prompt', '')[:200]}...")
+        
+        task_content = result.get('task_prompt_content', [])
+        logger.info(f"Task prompt content items: {len(task_content)}")
+        for i, item in enumerate(task_content[:3]):  # Log first 3 items
+            logger.info(f"Content item {i}: keys={list(item.keys())}")
+            if 'text' in item:
+                logger.info(f"  Text length: {len(item['text'])} characters")
+                logger.info(f"  Text sample (first 150 chars): {item['text'][:150]}...")
+            if 'image_uri' in item:
+                logger.info(f"  Image URI: {item['image_uri']}")
+        
+        if len(task_content) > 3:
+            logger.info(f"  ... and {len(task_content) - 3} more content items")
+
+        logger.debug(f"Complete result output: {json.dumps(result, indent=2)}")
+        logger.info("=== DYNAMIC FEW-SHOT LAMBDA COMPLETED ===")
+        return result
+
+    except Exception as e:
+        logger.error(f"=== DYNAMIC FEW-SHOT LAMBDA ERROR ===")
+        logger.error(f"Error type: {type(e).__name__}")
+        logger.error(f"Error message: {str(e)}")
+        logger.error(
+            f"Input event keys: {list(event.keys()) if 'event' in locals() else 'Unknown'}"
+        )
+        # In demo, we'll fail gracefully with detailed error info
+        raise Exception(f"Dynamic few-shot Lambda failed: {str(e)}")
+
+def _handle_input_document(placeholders, default_system_prompt, default_task_prompt):
+    """
+    Handle input request and return custom system_prompt and task_prompt_content
+    """
+    substitutions = {
```

Comment:
Why not include DOCUMENT_IMAGE in substitutions?  Not sure why images are any different than text in the substitutions?  The code would be cleaner if you treat them the same.